tensorflow/python/framework/func_graph.py

```diff
@@ -187,7 +187,7 @@ def __init__(self, name, collections=None, capture_by_value=None):
     self.inputs = []
     self.outputs = []
     self.control_outputs = []
-    self.control_captures = set()
+    self.control_captures = object_identity.ObjectIdentitySet()
```

Comment:
Why is this change needed? We try to avoid using `ObjectIdentitySet` internally due to the performance overhead, so if possible it would be best to avoid using it.